test/operator/test_continuous_operator.py

```diff
@@ -100,22 +100,19 @@ def test_potential_energy():
 
 
 def test_kinetic_energy():
-    for dtype in (jnp.float32, jnp.float64):
```

Comment:
can we keep the old test around as well?

also your test is not asserting the dtypes.

Reply:
I moved all tests about dtype in this file to a new test `test_dtype` at the bottom of this file.

The previous test only varied the the dtype of `x`, not the operators. Now I vary both of them, and check that the dtype of energy is promoted from both of them.